src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -158,7 +175,27 @@ public virtual void SetPinMode(int pinNumber, PinMode mode)
                 throw new InvalidOperationException($"Pin {pinNumber} does not support mode {mode}.");
             }
 
-            _driver.SetPinMode(logicalPinNumber, mode);
+            if (_openPins.TryGetValue(pinNumber, out var desired) && desired.HasValue)
+            {
+                _driver.SetPinMode(logicalPinNumber, mode, desired.Value);
+            }
+            else
+            {
+                _driver.SetPinMode(logicalPinNumber, mode);
+            }
+        }
+
+        /// <summary>
+        /// Sets the mode to a pin and sets an initial value for an output pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        /// <param name="initialValue">The initial value if the <paramref name="mode"/> is output. The driver will do it's best to prevent glitches to the other value when
+        /// changing from input to output.</param>
+        public void SetPinMode(int pinNumber, PinMode mode, PinValue initialValue)
```

Comment:
mm I'm not so sure about this one. I'm fine with OpenPin having an initialValue as that makes sense always, but initialValue when changing a mode only makes sense when doing output. Since we allow to write on input pins now, I would expect folks wanting to have a immediate value would be calling Write() and then changing the mode from input to output.

Reply:
You are right, but I thought I would add the convenience method anyway (the overload is also suggested in #721)

Reply:
I agree with @joperezr on this. Since we allow to change value in Input mode I don't believe there is much benefit of doing this and it's confusing

Reply:
[Triage]: SetPinMode with initial value is just a convinience method so that you don't have to call Write() and then SetPinMode(). Given it is really only saving one call, It's better to remove this so as to not to overcomplicate the API on the controller.

Reply:
Removed overload, as discussed.